src/useObservable.ts

```diff
@@ -22,7 +21,13 @@ function getOrCreateStore<T>(inputObservable: Observable<T>, initialValue: T) {
       shareReplay({refCount: true, bufferSize: 1}),
       tap(value => (entry.currentValue = value)),
     )
+
+    // Eagerly subscribe to sync set `entry.currentValue` to what the observable returns
+    // @TODO: perf opt opportunity: don't setup sync subscription initialValue is !== undefined
+    // Why not check `initialValue`? Because it might change during re-renders, but here we're only concerned with the first run
+    // if (entry.currentValue === undefined) {
```

Comment:
I've tested this optimization and I didn't find any regressions.
However, given the scope of how it *potentially* impacts Studio v3 internals, I decided to comment it out for now.